keps/sig-scheduling/5471-enable-sla-based-scheduling/README.md

```diff
@@ -0,0 +1,855 @@
+
+- **No Implicit Conversion**: Values like `"0950"` vs `"950"` are numerically equal but may confuse users expecting string matching behavior.
+
+- **Parsing Overhead**: Each taint/toleration match with numeric operators requires integer parsing.
+
+- Invalid taints meant to be used with the new comparison operators (e.g., `node.kubernetes.io/sla=95.5` and `node.kubernetes.io/version=1`) are not detected at admission time.
+
+- **Taint Misconfiguration Risk**: When nodes have taints with non-numeric values (e.g., `node.kubernetes.io/sla=high` instead of `node.kubernetes.io/sla=950`) that are intended for use with numeric operators, the misconfiguration is only detected during pod scheduling attempts, not at taint creation time. This can lead to scheduling failures that are difficult to diagnose.
+
+### Risks and Mitigations
+
+#### Scheduler Performance Regression
+
+**Risk**: Integer parsing during taint/toleration matching could degrade scheduler performance, especially in clusters with thousands of taints.
+
+**Mitigation**:
+
+- Parse integers only when new operators are used.
+- Existing `Equal`/`Exists` operators execute identical code paths with no additional overhead.
+- Consider caching parsed values in scheduler data structures if performance issues arise
+- Feature gate allows disabling if performance problems occur
+
+#### API Compatibility and Version Skew
+
+**Risk**: Pods using new operators cannot be scheduled if some schedulers don't support the feature, creating deployment failures during upgrades.
+
+**Mitigation**:
+
+- Feature gate prevents usage until all components are upgraded
+- Clear upgrade documentation specifying component upgrade order
+- Backward compatibility testing ensures existing workloads continue functioning
+- Gradual rollout recommendations for production clusters
+
+#### Edge Cases in Numeric Parsing
+
+**Risk**: Unexpected behavior with edge cases like integer overflow, leading zeros, or malformed input could cause scheduling failures.
+
+**Mitigation**:
+
+- Use Go's standard `strconv.ParseInt()` with well-defined error handling
+- Comprehensive unit tests covering edge cases (overflow, underflow, malformed strings)
+- API validation rejects pods with unparseable values rather than silently failing
+- Clear error messages help users identify and fix configuration issues
+
+#### Taint Misconfiguration Detection
+
+**Risk**: Node taints intended for numeric comparison may contain non-numeric values (e.g., `node.kubernetes.io/sla=high` instead of `node.kubernetes.io/sla=950`), causing scheduling failures that are only detected during pod placement attempts rather than at taint creation time.
+
+**Mitigation**:
+
+- Clear documentation and examples showing proper numeric taint configuration
+- Enhanced error messages in scheduling events that clearly indicate parsing failures
```

Comment:
Plus we also need logs as well. because if only some nodes have non-numeric taint values, a pod could still go to other nodes with numeric taint values, which might result in cluster admins not noticing the issues specific on some nodes only.